object-construction-checker/src/main/java/org/checkerframework/checker/objectconstruction/ObjectConstructionAnnotatedTypeFactory.java

```diff
@@ -424,13 +424,16 @@ public boolean isSubtype(final AnnotationMirror subAnno, final AnnotationMirror
 
       if (AnnotationUtils.areSameByClass(subAnno, CalledMethodsPredicate.class)) {
         if (AnnotationUtils.areSameByClass(superAnno, CalledMethodsPredicate.class)) {
```

Comment:
This does not seem to handle the case of comparing a `@CalledMethods` annotation with a `@CalledMethodsPredicate` annotation.  Please see the tests I added.

Reply:
You're right, it didn't. I added that case by turning the `@CalledMethods` annotation into a conjunct and passing it to the solver.